scripts/gen_vimdoc.py

```diff
@@ -719,6 +736,11 @@ def para_as_map(parent, indent='', width=text_width - indentation, fmt_vimhelp=F
                 elif kind == 'warning':
                     text += render_node(child, text, indent=indent,
                                         width=width, fmt_vimhelp=fmt_vimhelp)
+                elif kind == 'since':
+                    since_match = re.match(r'^(\d+)', get_text(child))
+                    since = int(since_match.group(1)) if since_match else 0
+                    if since > NVIM_API_LEVEL or (since == NVIM_API_LEVEL and NVIM_API_PRERELEASE):
```

Comment:
if a "." is in the `@since` value we should compare with the Nvim version rather than api-level

Reply:
I don't think that's a good idea. We should pick one versioning scheme and stick with it; everything else will just lead to confusion.

Reply:
Yeah let's keep it consistent.

Would picking SemVer require moving the C stuff to SemVer as well?

Could be worth getting @bfredl's comment on the record about the merits of simple versioning (single number)

Reply:
> Would picking SemVer require moving the C stuff to SemVer as well?

Yes, which would be breaking. (Originally, the API level was intended to move much more slowly than the Neovim minor version.)

Reply:
> We should pick one versioning scheme and stick with it

How? The API level is versioned independently from Nvim version. Are we planning to increment API level when we add a new Lua function?

Reply:
Yes, that would be the implication. 

For the record: when I say "we should pick one versioning scheme", I meant "for Lua API". @bfredl suggested (re)using the Nvim API for that, since we already have tooling and documentation in place (we tend to prefer leveraging existing stuff over reinventing the wheel, after all), but there may be a good argument for using the Neovim version instead. (E.g., one argument I just thought of is that we already use Neovim version for deprecation, so it might make sense to use it for introduction as well.)

But no matter which we pick, I think we should _not_ allow _both_ `@since 12` and `@since 0.12` **for Lua functions**. (For Nvim API functions, it's `@since 12` or nothing.)

Reply:
> I think we should _not_ allow _both_ `@since 12` and `@since 0.12` **for Lua functions**.

Yes, of course. My suggestion was only for Lua functions.

Reply:
But wouldn't your suggestion allow both?

Reply:
> one argument I just thought of is that we already use Neovim version for deprecation, so it might make sense to use it for introduction as well.

`vim.deprecate` messages only last for a version, so SemVer-based deprecation messages can probably also be phased out in a version or two

Reply:
gen_vimdoc.py should throw an error if the wrong version format is used depending on the function kind (API/Lua).

Reply:
> gen_vimdoc.py should throw an error if the wrong version format is used depending on the function kind (API/Lua).

Yes, that's exactly what I was getting at.